scripts/setfit/run_fewshot_distillation.py

```diff
@@ -245,18 +236,60 @@ def train(self):
                 # if os.path.exists(results_path):
                 #     continue
 
-                if (self.mode == self.TEACHER) or (self.mode == self.SETFIT_STUDENT):
-                    self.model.load_state_dict(copy.deepcopy(self.model_wrapper.model_original_state))
-                    metrics = self.train_eval_setfit(
-                        fewshot_ds[name],
-                        test_dataset,
-                        self.loss_class,
-                        self.args.num_epochs,
-                        metric,
+                if self.mode == self.TEACHER:
+                    # self.model.load_state_dict(copy.deepcopy(self.model_wrapper.model_original_state))
+                    fewshot_ds[name].rename_column(
+                        "text", "sentence"
+                    )  # Map dataset columns to sentence/label expected by trainer
+                    teacher_model = SetFitModel.from_pretrained(self.model_name)
+                    teacher_trainer = SetFitTrainer(
+                        model=teacher_model,
+                        train_dataset=fewshot_ds[name],
+                        eval_dataset=eval_dataset,
+                        loss_class=CosineSimilarityLoss,
```

Comment:
Should we add the type with something like this?

```suggestion
                        loss_class: torch.nn.Module =CosineSimilarityLoss,
```

Reply:
good point. done